common/beerocks/btl/btl.cpp

```diff
@@ -41,16 +41,23 @@ void transport_socket_thread::set_select_timeout(unsigned msec) { poll_timeout_m
 bool transport_socket_thread::send_cmdu_to_bus(ieee1905_1::CmduMessageTx &cmdu_tx,
                                                const std::string &dst_mac,
                                                const std::string &src_mac)
+{
+    return send_cmdu_to_bus(cmdu_tx, 0, dst_mac, src_mac);
+}
+
+bool transport_socket_thread::send_cmdu_to_bus(ieee1905_1::CmduMessageTx &cmdu_tx,
+                                               uint32_t iface_index, const std::string &dst_mac,
+                                               const std::string &src_mac)
 {
     if (!cmdu_tx.finalize()) {
         THREAD_LOG(ERROR) << "finalize failed";
         return false;
     }
 
-    return send_cmdu_to_bus(cmdu_tx, dst_mac, src_mac, cmdu_tx.getMessageLength());
+    return send_cmdu_to_bus(cmdu_tx, iface_index, dst_mac, src_mac, cmdu_tx.getMessageLength());
 }
 
-bool transport_socket_thread::send_cmdu_to_bus(ieee1905_1::CmduMessage &cmdu,
+bool transport_socket_thread::send_cmdu_to_bus(ieee1905_1::CmduMessage &cmdu, uint32_t iface_index,
```

Comment:
I don't understand it.
The message is sent to the bus with src mac which is the bridge/al_mac.
doesn't ifindex means index of radio interface? Because we send the message from the "bridge" interface. Does it mean face index of the bridge maybe? It is not clear to me.
Anyway, I don't think that it is a good idea to add it as an extra argument. For whatever reason we need it, couldn't it be generated internally on the function?

Reply:
According to the 1905.1 standard, we need to transmit some CMDUs (i.e. Topology Discovery message) once on each and every of the available 1905.1 interfaces. Thus why we have modified this method to include which interface to use. In a following commit we call this method in a loop for each bridge interface.

Please see description of the issue #1160 for further details

Reply:
I think @morantr's question was: shouldn't the source address be the interface address rather than the AL-MAC? The answer is: the IEEE1905.1 standard does allow the interface address to be used as the source address, but prefers the AL-MAC address.